### PR TITLE
Updated for certificate directory support

### PR DESCRIPTION
**Pull request by AssemblyJohn**
- added certificate bundle wrapper
- added general openssl types
- added some helpers
- updated certificate delete for bundle support

### CODE REVIEW

File: include/evse_utilities.hpp [hunk 1]
```diff
@@ -0,0 +1,66 @@
+
```
Comment (Pietfried) [inline include/evse_utilities.hpp:1]: License header missing

File: include/sec_types.hpp [hunk 2]
```diff
@@ -0,0 +1,60 @@
+#ifndef EVSE_SEC_TYPES_HPP
```
Comment (Pietfried) [inline include/sec_types.hpp:1]: License header missing

File: include/x509_bundle.hpp [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    std::string to_export_string() const;
+
+    /// @brief Exports the full certificate chain either as individual files if it is using a directory
+    /// or as a bundle if it uses a bundle file, at the initially provided path. Also deletes/adds the updated
+    /// certificates
+    /// @return true on success, false otherwise
+    bool export_certificates();
+
+    /// @brief Syncs the file structure with the certificate store adding certificates that are not found on the
+    /// storeage and deleting the certificates that are not contained in this bundle
```
Comment (barsnick) [inline include/x509_bundle.hpp:67]: storage

File: include/x509_wrapper.hpp [hunk 4]
```diff
@@ -20,19 +21,33 @@ class CertificateLoadException : public std::runtime_error {
...
+/// @brief Convenience wrapper around openssl X509 certificate
+class X509Wrapper {
 public:
-    // Constructors
-    X509Wrapper(const std::string& certificate, const EncodingFormat encoding);
-    X509Wrapper(const std::filesystem::path& path, const EncodingFormat encoding);
+    X509Wrapper(const std::filesystem::path& file, const EncodingFormat encoding);
+    X509Wrapper(const std::string& data, const EncodingFormat encoding);
 
     /// @brief Since it implies ownership full transfer, must be very carefull with this that's why it's explicit
```
Comment (barsnick) [inline include/x509_wrapper.hpp:43]: carefull -> careful

File: include/evse_utilities.hpp [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+        return false;
+    }
+
+    static bool write_to_file(const std::filesystem::path& file_path, const std::string& data,
+                              std::ios::openmode mode) {
+        try {
+            std::ofstream fs(file_path, mode | std::ios::binary);
+            if (!fs.is_open()) {
+                // EVLOG_error << "Error opening file: " << file_path;
```
Comment (Pietfried) [inline include/evse_utilities.hpp:39]: Please uncomment or delete

**Review by Pietfried — APPROVED** [review]

**Pull request merged**